### PR TITLE
Eliminate one shared static this from thread.d

### PR DESCRIPTION
**Pull request by andralex**
The Windows implementation does not need any initialization.

### CODE REVIEW

**Comment by andralex** [comment]
Ah, spoke too fast. NetBSD is Posix so it uses the same implementation.

**Comment by andralex** [comment]
ping

**Comment by andralex** [comment]
ping

**Comment by CyberShadow** [comment]
CC @nrTQgc @nemanja-boric-sociomantic 

@andralex I think this PR would be easier to review if you described the changes in more detail (ideally in the commit message), as the diff is hard to make sense of even with ?w=1.

File: src/core/thread.d [hunk 1]
```diff
@@ -909,39 +909,159 @@ class Thread
...
+        {
+            Priority result;
+            version( Solaris )
+            {
+                pcparms_t pcParms;
+                pcinfo_t pcInfo;
 
+                pcParms.pc_cid = PC_CLNULL;
+                if (priocntl(idtype_t.P_PID, P_MYID, PC_GETPARMS, &pcParms) == -1)
+                    assert( 0, "Unable to get scheduling class" );
```
Comment (nemanja-boric-sociomantic) [inline src/core/thread.d]: Nit pick: these asserts could have the unified syntax with `version(Posix)` - no `if`s

File: src/core/thread.d [hunk 2]
```diff
@@ -909,39 +909,159 @@ class Thread
...
+        private struct Priority
+        {
+            int PRIORITY_MIN = int.min;
+            int PRIORITY_DEFAULT = int.min;
+            int PRIORITY_MAX = int.min;
+        }
 
+        static int loadGlobal(string which)() //@nogc nothrow pure @safe
+        {
+            shared Priority cache;
```
Comment (nemanja-boric-sociomantic) [inline src/core/thread.d]: Can this be `static shared Priority cache`?

**Comment by nemanja-boric-sociomantic** [comment]
Other than that LGTM.

**Comment by nemanja-boric-sociomantic** [comment]
Just to check, this comment:

```
The NetBSD version was elaborately just returning zero it seems, so I simplified that
```

is not valid anymore, right? (I can't see any changes done for `NetBSD`, which your second comment seems to confirm).

**Comment by Burgos** [comment]
Ping :-)

File: src/core/thread.d (hunk 2 above)
Comment (andralex) [inline src/core/thread.d]: @nemanja-boric-sociomantic ah, that was the whole idea :). Thanks!

File: src/core/thread.d (hunk 1 above)
Comment (andralex) [inline src/core/thread.d]: ok

**Comment by andralex** [comment]
@nemanja-boric-sociomantic yah, that comment about NetBSD is not valid, sorry for the confusion.

File: src/core/thread.d (hunk 1 above)
Comment (andralex) [inline src/core/thread.d]: Looking over code, I don't get the idea for refactoring.

**Comment by andralex** [comment]
@nemanja-boric-sociomantic done

**Comment by andralex** [comment]
@CyberShadow added comments

File: src/core/thread.d (hunk 1 above)
Comment (Burgos) [inline src/core/thread.d]: Yeah, me neither anymore. All good, sorry.

Comment (Burgos) [inline src/core/thread.d]: Ah, I see it now: https://github.com/dlang/druntime/pull/1837/files#diff-8bb12ed976acf0a5132e877ec5a01ea8R1021

`priocntl(...) != -1 || assert (0, "Unable to get scheduling class.");`

I don't think it's important, though.

**Comment by Burgos** [comment]
Looks good, thanks!

**Comment by nemanja-boric-sociomantic** [comment]
Just needs squash @andralex

**Pull request merged**